.github/workflows/osx-sccache.yml

```diff
@@ -0,0 +1,52 @@
+name: osx-sccache
+
+on:
+  push:
+    branches: [master, release-*, gha-test-*]
```

Comment:
The turnaround time of the job is ~1hr (https://github.com/diem/diem/runs/1864797013?check_suite_focus=true), and the land time is ~0.5 hr.  On a heavy day of >15 PRs, this sccache job would be lagging behind master by hours.  

Have we thought about the tradeoff in generating the cache artifacts daily/nightly?

Reply:
Indeed, too much of this time is spent building tools as well....    debating making it every few hours.   but during the ramp up time where people test it out I think it's valuable to have it fully up to date before dropping every 2-3 hours if changes exist.   1/3 of the time is building the tooling.

Reply:
we can land this now and iterate.